lib/ioutil/log_reader.go

```diff
@@ -44,34 +41,37 @@ func NewLogReader(r io.ReadCloser) *LogReader {
 // a LogEntry struct, and then writes that struct's log message to the
 // supplied []byte slice.
 func (lr *LogReader) Read(p []byte) (int, error) {
+	// use unwritten previous message if we happen to have one
+	msg := lr.prev
 
-	entry := &LogEntry{}
-
-	if lr.prev != nil {
-		entry = lr.prev
+	// scanning one Read call at a time allows us to stay current when tailing
+	// output from the underlying stream
+	if msg == nil && lr.s.Scan() {
+		l := &jlexer.Lexer{Data: lr.s.Bytes()}
+		entry := &LogEntry{}
+		entry.UnmarshalEasyJSON(l)
+		msg = &entry.Msg
 	} else {
-		if lr.s.Scan() {
-
-			l := &jlexer.Lexer{Data: lr.s.Bytes()}
-			entry.UnmarshalEasyJSON(l)
-
-		} else {
-			err := lr.s.Err()
-			if err == nil {
-				return 0, io.EOF
-			}
+		// If we get here, there are no more bytes to scan from the underlying stream.
```

Comment:
What happens if msg != nil? In that case wouldn't you want to do continue at line 65? With the code as is, you will go to the else clause and return either EOF or whatever error Err() returns.